src/chai/ManagedArray.inl

```diff
@@ -378,21 +377,21 @@ template<bool B, typename std::enable_if<B, int>::type>
 CHAI_INLINE
 CHAI_HOST_DEVICE
 void
-ManagedArray<T>::moveInnerData()
+ManagedArray<T>::moveInnerImpl()
 {
-   for (int i = 0; i < size(); ++i)
-   {
-      // Copy constructor triggers data movement
-      T temp = T(m_active_pointer[i]);
-   }
+  // Use reverse order to optimize cnmem deallocation
```

Comment:
I don't think this comment should be here anymore 😄

Reply:
You don't think it needs to be in reverse order anymore?

Reply:
Fixed!